WallsAndHoles/tilemap.cpp

```diff
@@ -171,6 +171,26 @@ void TileMap::removingTileTemplateSet(TileTemplateSet *tileTemplateSet)
     mPingingMutex.unlock();
 }
 
+void TileMap::removingTileTemplate(TileTemplate *tileTemplate)
+{
+    if (!tileTemplate) return;
+
+    mPingingMutex.lock();
```

Comment:
I recently learned that using a QMutexLocker is a really good idea. It keeps track of a QMutex and unlocks it in the destructor. If you make an auto-storage variable like so
```
QMutexLocker locker(&mPingingMutex)
```
then the mutex is guaranteed to be unlocked when the method exists.